docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+    events where the object content is not modified) and then creates a
+    key to store a reference to the object in the work queue. The key
+    contains the GVK key used to retrieve a reference to the informer
+    and lister for that kind of object, and a namespace/name key to
+    retrieve the actual object. Storing the key in the work queue is a
+    common pattern in client-go as the object may have changed in the
+    cache (which is kept updated by the reflector) by the time a worker
+    gets a copy from the work queue. Workers should always receive the
+    key and use it to retrieve the object from the cache.
+
+3.  A worker pulls a key from the work queue, and then does the
+    following processing:
+
+    -  Uses the GVK key to get the reference to the lister for the
+        object
+    -  Gets the object from the lister cache using the key that was
+        generated with the MetaNamespaceKeyFunc.
+    -  Gets the lister for Placement objects and list all placements
+    -  Iterates on all placements, and for each placement:
+        - evaluates the placement label selector vs. object labels to
+          check if there is a match.
+        - If there is a match, uses the lister for ManagedCluster and
+          find the matching clusters using the label selector expression
+          for clusters.
+        - If there are matching clusters, add the names of the cluster
+          to a hashmap setting the name of the cluster as a key. Cluster
+          groups from different placements are merged together.
+    -  If there are no matching clusters, the worker returns without
+        actions and is ready to process other events from the queue.
+    -  If there are matching clusters:
+        - Wraps the object into a ManifestWork
+        - Adds a label to the ManifestWork that is used to track the
+          placement that caused the object to be delivered to one or
+          more clusters. The label contains both the placement name
+          (note that placement is cluster-scoped) and the WDS name. A
+          label is attached for each placement whose selector matches
+          the object. This way, when a placement is deleted or
+          updated it is possible to locate the associated ManifestWorks
+          for deletion or label removal (if more than one label is
+          present, as other placements may “own” the object).
+        - For each cluster:
+          -  Sets the manifest namespace == name of the cluster
+          -  Uses the client connected to the ITS to do a server-side
+              apply of the manifest into the namespace.
+        - Worker returns and is ready to pick other keys from the queue.
+
+There are other event flows, based on the object GVK and type of event.
+The following sections broadly describe these flows:
+
+#### Object Deleted 
+
+When an object is deleted from the WDS, the handler’s *DeleteFunc* is
+invoked. Since most objects do not have a finalizer (which prevents
```

Comment:
What about a workload object that is deleted after one run on this controller and before the next?

Reply:
Is this question still related to the discussion on deleted and re-created object with same identifiers (lines 373-376) ? Can you elaborate more on the scenario?

Reply:
No, this question is about handling changes that result in no event notifications from the informer, because they happen during an interval of time when the controller is crashed.

Reply:
Ok, this is listed in https://github.com/kubestellar/kubestellar/issues/1674 item 2, so we can keep track there.